packages/core/src/domain/configuration.ts

```diff
@@ -200,33 +222,58 @@ export function buildCookieOptions(userConfiguration: UserConfiguration) {
+function getEndpoint(type: EndpointType, conf: TransportConfiguration, source?: string, withLegacyHost = false) {
   const tags =
     `sdk_version:${conf.sdkVersion}` +
     `${conf.env ? `,env:${conf.env}` : ''}` +
     `${conf.service ? `,service:${conf.service}` : ''}` +
     `${conf.version ? `,version:${conf.version}` : ''}`
-  const datadogHost = `${type}-http-intake.logs.${conf.site}`
+  const datadogHost = getHost(type, conf, withLegacyHost)
   const host = conf.proxyHost ? conf.proxyHost : datadogHost
   const proxyParameter = conf.proxyHost ? `ddhost=${datadogHost}&` : ''
   const applicationIdParameter = conf.applicationId ? `_dd.application_id=${conf.applicationId}&` : ''
   const parameters = `${applicationIdParameter}${proxyParameter}ddsource=${source || 'browser'}&ddtags=${tags}`
 
   return `https://${host}/v1/input/${conf.clientToken}?${parameters}`
 }
+function getHost(type: EndpointType, conf: TransportConfiguration, useLegacyDomain: boolean) {
+  if (!useLegacyDomain) {
+    const subdomains = {
+      [EndpointType.BROWSER]: 'logs',
+      [EndpointType.RUM]: 'rum',
+      [EndpointType.TRACE]: 'trace',
+    }
+    if (conf.site === INTAKE_SITE[Datacenter.US]) {
+      return `${subdomains[type]}.browser-intake-datadoghq.com`
+    }
+    if (conf.site === STAGING_INTAKE_SITE[Datacenter.US]) {
+      return `${subdomains[type]}.browser-intake-datad0g.com`
+    }
+  }
+  return `${type}-http-intake.logs.${conf.site}`
+}
 
 export function isIntakeRequest(url: string, configuration: Configuration) {
-  return (
-    getPathName(url).indexOf('/v1/input/') !== -1 &&
-    (haveSameOrigin(url, configuration.logsEndpoint) ||
-      haveSameOrigin(url, configuration.rumEndpoint) ||
-      haveSameOrigin(url, configuration.traceEndpoint) ||
-      (!!configuration.internalMonitoringEndpoint && haveSameOrigin(url, configuration.internalMonitoringEndpoint)) ||
-      (!!configuration.replica &&
-        (haveSameOrigin(url, configuration.replica.logsEndpoint) ||
-          haveSameOrigin(url, configuration.replica.rumEndpoint) ||
-          haveSameOrigin(url, configuration.replica.internalMonitoringEndpoint))))
-  )
+  if (getPathName(url).indexOf('/v1/input/') !== -1) {
+    const origins = getIntakeEndpoints(configuration).map(getOrigin)
+    return origins.indexOf(getOrigin(url)) !== -1
+  }
+  return false
+}
+
+function getIntakeEndpoints(configuration: Configuration) {
+  const endpoints = [configuration.logsEndpoint, configuration.rumEndpoint, configuration.traceEndpoint]
+  if (!!configuration.internalMonitoringEndpoint) {
+    endpoints.push(configuration.internalMonitoringEndpoint)
+  }
+  if (!!configuration.replica) {
+    endpoints.push(
+      configuration.replica.logsEndpoint,
+      configuration.replica.rumEndpoint,
+      configuration.replica.internalMonitoringEndpoint
+    )
+  }
+  return endpoints.concat(configuration.legacyEndpoints || [])
```

Comment:
I think we could simplify the problem by not checking explicitly every possible endpoints but only if host match:
- `-http-intake.logs.${conf.site}`
- `.browser-intake-${conf.site}`
- proxyHost

Reply:
In order to do that we need to access the `site` from `isIntakeRequest`
```
function isIntakeRequest(url: string, configuration: Configuration)
```
One way would be to add `site` to the `Configuration` type. What do you think?

Reply:
I agree.

Similar to your current implementation, we could clarify the intent by computing and storing an `intakeDomains` array instead.